src/App.tsx

```diff
@@ -1,56 +1,92 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownList } from './Components/DrowdownList/DrowdownList';
+
+const debounce = <T extends (...args: never[]) => void>(
+  func: T,
+  delay: number,
+): ((...args: Parameters<T>) => void) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: Parameters<T>) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [searchQuery, setSearchQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isVisibleDropdown, setIsVisibleDropdown] = useState(false);
+  const [isVisibleSuggestion, setIsVisibleSuggestion] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [appliedQuery],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(person => {
+      setIsVisibleSuggestion(true);
+
+      const nameNormalize = person.name.toLowerCase();
+      const searchQueryNormalize = appliedQuery.toLowerCase().trim();
+
+      return nameNormalize.includes(searchQueryNormalize);
+    });
+  }, [peopleFromServer, appliedQuery]);
```

Comment:
Move `setState` from iteration

```suggestion
  const visiblePeople = useMemo(() => {
    setIsVisibleSuggestion(true);
  
    return peopleFromServer.filter(person => {
      const nameNormalize = person.name.toLowerCase();
      const searchQueryNormalize = appliedQuery.toLowerCase().trim();

      return nameNormalize.includes(searchQueryNormalize);
    });
  }, [peopleFromServer, appliedQuery]);
```